src/main/java/com/gradle/enterprise/bamboo/config/PluginEnabledListener.java

```diff
@@ -0,0 +1,47 @@
+package com.gradle.enterprise.bamboo.config;
+
+import com.atlassian.bamboo.bandana.PlanAwareBandanaContext;
+import com.atlassian.bandana.BandanaManager;
+import com.atlassian.event.api.EventListener;
+import com.atlassian.plugin.event.events.PluginEnabledEvent;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class PluginEnabledListener {
```

Comment:
Maybe something like

```suggestion
public class ConfigurationMigrator {
```

so the purpose of the component is more clear?

Reply:
I think we can easily write a unit test for this class. 2 tests will be enough. One to check that filtering is working and one for the migration logic.